CollectNewspaperKarel.java

```diff
@@ -1,17 +1,50 @@
-/*
- * File: CollectNewspaperKarel.java
- * --------------------------------
- * At present, the CollectNewspaperKarel subclass does nothing.
- * Your job in the assignment is to add the necessary code to
- * instruct Karel to walk to the door of its house, pick up the
- * newspaper (represented by a beeper, of course), and then return
- * to its initial position in the upper left corner of the house.
- */
-
-import stanford.karel.*;
-
-public class CollectNewspaperKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: CollectNewspaperKarel.java
+ * --------------------------------
+ * At present, the CollectNewspaperKarel subclass does nothing.
+ * Your job in the assignment is to add the necessary code to
+ * instruct Karel to walk to the door of its house, pick up the
+ * newspaper (represented by a beeper, of course), and then return
+ * to its initial position in the upper left corner of the house.
+ */
+
+import stanford.karel.*;
+
+public class CollectNewspaperKarel extends SuperKarel {

```

Comment:
Your code works but to make it easy to understand it would have been good if you had broken it down into methods. So the run function would have been something like this:

```
public void run() {
    pickNewsPaper();
    returnToStartingPoint();
}
```